SU2_CFD/include/fluid/CCoolProp.hpp

```diff
@@ -0,0 +1,146 @@
+        su2double Gamma{0.0};           /*!< \brief Ratio of Specific Heats. */
+        su2double Gas_Constant{0.0};    /*!< \brief specific Gas Constant. */
+        su2double Pressure_Critical{0.0};   /*!< \brief critical pressure */
+        su2double Temperature_Critical{0.0};    /*!< \brief critical temperature */
+#ifdef USE_COOLPROP
+        std::unique_ptr<CoolProp::AbstractState> fluid_entity;   /*!< \brief fluid entity */
+#endif
+
+public:
+/*!
+ * \brief Constructor of the class.
+ */
+CCoolProp(string fluidname);
+
+#ifdef USE_COOLPROP
+/*!
+ * \brief Set the Dimensionless State using Density and Internal Energy
+ * \param[in] rho - first thermodynamic variable.
+ * \param[in] e - second thermodynamic variable.
+ */
+void SetTDState_rhoe(su2double rho, su2double e) override;
+
+/*!
+ * \brief Set the Dimensionless State using Pressure  and Temperature
+ * \param[in] P - first thermodynamic variable.
+ * \param[in] T - second thermodynamic variable.
+ */
+void SetTDState_PT(su2double P, su2double T) override;
+
+/*!
+ * \brief Set the Dimensionless State using Pressure and Density
+ * \param[in] P - first thermodynamic variable.
+ * \param[in] rho - second thermodynamic variable.
+ */
+void SetTDState_Prho(su2double P, su2double rho) override;
+
+/*!
+ * \brief Set the Dimensionless Internal Energy using Pressure and Density
+ * \param[in] P - first thermodynamic variable.
+ * \param[in] rho - second thermodynamic variable.
+ */
+void SetEnergy_Prho(su2double P, su2double rho) override;
+
+/*!
+ * \brief Set the Dimensionless State using Enthalpy and Entropy
+ * \param[in] th1 - first thermodynamic variable (h).
+ * \param[in] th2 - second thermodynamic variable (s).
+ *
+ */
+void SetTDState_hs(su2double h, su2double s) override;
+
+/*!
+ * \brief Set the Dimensionless State using Density and Temperature
+ * \param[in] th1 - first thermodynamic variable (rho).
+ * \param[in] th2 - second thermodynamic variable (T).
+ *
+ */
+void SetTDState_rhoT(su2double rho, su2double T) override;
+
+/*!
+ * \brief Set the Dimensionless State using Pressure and Entropy
+ * \param[in] th1 - first thermodynamic variable (P).
+ * \param[in] th2 - second thermodynamic variable (s).
+ */
+void SetTDState_Ps(su2double P, su2double s) override;
+
+/*!
+ * \brief compute some derivatives of enthalpy and entropy needed for subsonic inflow BC
+ * \param[in] InputSpec - Input pair for FLP calls ("Pv").
+ * \param[in] th1 - first thermodynamic variable (P).
+ * \param[in] th2 - second thermodynamic variable (v).
+ *
+ */
+void ComputeDerivativeNRBC_Prho(su2double P, su2double rho) override;
+#endif
+
+/*!
+     * \brief Get the value of the critical pressure.
+     * \return Critical pressure.
+*/
+su2double GetPressure_Critical(void) const { return Pressure_Critical; }
+
+/*!
+     * \brief Get the value of the critical temperature.
+     * \return Critical temperature.
+*/
+su2double GetTemperature_Critical(void) const { return Temperature_Critical; }
```

Comment:
please fix this formatting

Reply:
Hi, I am sorry that I do not get your idea.

Could you point out the issue of the formatting?

Reply:
Have you run clang-format on new files?